Makefile

```diff
@@ -129,9 +129,7 @@ test: ## Run the unit tests for the codebase.
 # This runs the unit tests for the codebase with coverage check.
 .PHONY: test-coverage
 test-coverage: ## Run the unit tests for the codebase with coverage check.
-	@mkdir -p $(OUTPUT_DIR)
-	@$(MAKE) test GO_TEST_ARGS="-coverprofile=$(OUTPUT_DIR)/go-test-coverage.out -covermode=atomic -coverpkg=github.com/envoyproxy/ai-gateway/... $(GO_TEST_ARGS)"
-	@go tool go-test-coverage --config=.testcoverage.yml
+	@$(MAKE) test GO_TEST_ARGS="-coverprofile=coverage.xml -covermode=atomic -coverpkg=github.com/envoyproxy/ai-gateway/... $(GO_TEST_ARGS)"
```

Comment:
help me understand this correctly but this itself is not failing the CI right? How can a contributor know if the current change passes all the coverage checks without pushing it? That's why I used go-test-coverage in the first place, and i wonder if it's possible to do so with codecov setting.

Reply:
codecov does not show full project coverage

![image](https://github.com/user-attachments/assets/a36df9a9-61c5-4c96-ba5f-beff7eaccfdf)